File: javacode/src/test/java/com/example/AddTest.java
package com.example;

import static org.junit.Assert.assertEquals;
import org.junit.Test;

public class AddTest {
  @Test
  public void testAdd1() {
    Add add = new Add(new Literal(5), new Literal(42));
    assertEquals(47, add.evaluate());
  }

  @Test
  public void testAdd2() {
    Add add = new Add(new Add(new Literal(5), new Literal(6)), new Literal(42));
    assertEquals(53, add.evaluate());
  }

  @Test
  public void testAdd3() {
    Add add = new Add(new Literal(42), new Add(new Literal(5), new Literal(6)));
    assertEquals(53, add.evaluate());
  }

  @Test
  public void testAdd4() {
    Add add = new Add(new Add(new Literal(5), new Literal(6)), new Add(new Literal(5), new Literal(6)));
    assertEquals(22, add.evaluate());
  }
}

File: csharpcode/ExpressionEvaluator.Tests/AddTest.cs
namespace ExpressionEvaluator.Tests;

public class AddTests
{
    [Test]
    public void TestAdd1()
    {
        Add add = new Add(new Literal(5), new Literal(42));
        Assert.That(add.Evaluate(), Is.EqualTo(47));
    }

    [Test]
    public void TestAdd2()
    {
        Add add = new Add(new Add(new Literal(5), new Literal(6)), new Literal(42));
        Assert.That(add.Evaluate(), Is.EqualTo(53));
    }
    [Test]
    public void TestAdd3()
    {
        Add add = new Add(new Literal(42), new Add(new Literal(5), new Literal(6)));
        Assert.That(add.Evaluate(), Is.EqualTo(53));
    }

    [Test]
    public void TestAdd4()
    {
        Add add = new Add(new Add(new Literal(5), new Literal(6)), new Add(new Literal(5), new Literal(6)));
        Assert.That(add.Evaluate(), Is.EqualTo(22));
    }
}
